### PR TITLE
Fix RemoteVideoUpdate usage on reconnect

### PR DESCRIPTION
**Pull request by hensmi-amazon**
**Issue #:** https://github.com/aws/amazon-chime-sdk-js/issues/2560

**Description of changes:** Fix for reported issue, which was due to not resetting `lastVideosToReceive`. Took the oppurtunity to do some refactoring, unfortunately its hard to enforce that 'Note' in the docstring for `AudioVideoControllerState` but the function now existing in the same file for most resetting logic may make it harder to miss.

**Testing:**

No longer can reproduce issues. Tested 4 person simulcast calls.

*Can these tested using a demo application? Please provide reproducible step-by-step instructions.*
1. Have 4 attendees join with simulcast and video enabled
2. Have one attendee disconnect wifi, and reconnect after 10 seconds.
3. Confirm start of receive video again, and start of send video as well.
4. Repeat multiple times, as this is not a 100% repro

**Checklist:**

1. Have you successfully run `npm run build:release` locally?
Y

5. Do you add, modify, or delete public API definitions? If yes, has that been reviewed and approved?
y

6. Do you change the wire protocol, e.g. the request method? If yes, has that been reviewed and approved?
y

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: demos/browser/app/meetingV2/video/PaginationManager.ts [hunk 1]
```diff
@@ -28,7 +28,7 @@ export default class PaginationManager<Type> {
     if (index === -1) {
       return;
     }
-    this.all.splice(index);
+    this.all.splice(index,1);
```
Comment (hensmi-amazon) [inline demos/browser/app/meetingV2/video/PaginationManager.ts:31]: Unrelated demo fix.

File: src/audiovideocontroller/DefaultAudioVideoController.ts [hunk 2]
```diff
@@ -813,9 +816,6 @@ export default class DefaultAudioVideoController
     }
     this.sessionStateController.perform(SessionStateControllerAction.FinishDisconnecting, () => {
       if (!reconnecting) {
-        // Do a hard reset of the signaling client in case this controller is reused;
-        // this will also cause `this.meetingSessionContext` to be reset if reused.
-        this.meetingSessionContext.signalingClient = null;
```
Comment (hensmi-amazon) [inline src/audiovideocontroller/DefaultAudioVideoController.ts:818]: Consolidating on the cleanup functions. When i added this to work around stale values of `videoSubscriptions` i didn't realize those tasks existed....

File: src/task/AttachMediaInputTask.ts [hunk 3]
```diff
@@ -46,7 +46,6 @@ export default class AttachMediaInputTask extends BaseTask {
           VideoLogEvent.InputAttached,
           this.context.videoDeviceInformation
         );
-        this.context.videoInputAttachedTimestampMs = Date.now();
```
Comment (hensmi-amazon) [inline src/task/AttachMediaInputTask.ts:49]: Unused

File: src/audiovideocontroller/AudioVideoControllerState.ts [hunk 4]
```diff
@@ -162,4 +171,63 @@ export default class AudioVideoControllerState {
...
+    this.previousSdpOffer = null;
+    this.sdpOfferInit = null;
+
+    // `audioVideoController` members should either be reusable, or moved to `AudioVideoControllerState` and
+    // cleaned up here.
+
+    // We don't want to mutate `videoTileController` as most video tiles will still be there on reconnect. We can remove tiles on the
+    // first index we receive if they no longer exist
+
+    // `mediaStreamBroker`, `activeAudioInput`, and `activeVideoInput` are cleaned up seperately in `ReleaseMediaStreamsTask`
```
Comment (ltrung) [inline src/audiovideocontroller/AudioVideoControllerState.ts:204]: Didn't `ReleaseMediaStreamTask` no longer exists?

Comment (hensmi-amazon) [reply src/audiovideocontroller/AudioVideoControllerState.ts:204]: Hah, you are right, I must have been looking at old branches. Let me fix up this comment

**Review by ltrung — APPROVED** [review]

File: src/audiovideocontroller/DefaultAudioVideoController.ts [hunk 5]
```diff
@@ -294,7 +294,7 @@ export default class DefaultAudioVideoController
   }
 
   private uninstallPreStartObserver(): void {
-    this.meetingSessionContext.signalingClient?.removeObserver(this.preStartObserver);
+    this.meetingSessionContext.signalingClient.removeObserver(this.preStartObserver);
```
Comment (hensmi-amazon) [inline src/audiovideocontroller/DefaultAudioVideoController.ts:297]: Undoes https://github.com/aws/amazon-chime-sdk-js/pull/2299/files which is no longer needed due to removal below. Not possible to unit test (but the previous test added still exists)

**Review by ltrung — APPROVED** [review]
